saleor/payment/gateways/authorize_net/plugin.py

```diff
@@ -0,0 +1,188 @@
+            "help_text": ("Provide private Authorize.Net Transaction Key."),
+            "label": "Transaction Key",
+        },
+        "client_key": {
+            "type": ConfigurationTypeField.STRING,
+            "help_text": ("Provide public Authorize.Net Client Key."),
+            "label": "Client Key",
+        },
+        "use_sandbox": {
+            "type": ConfigurationTypeField.BOOLEAN,
+            "help_text": "Determines if Saleor should use Authorize.Net sandbox environment.",
+            "label": "Use sandbox",
+        },
+        "store_customers_card": {
+            "type": ConfigurationTypeField.BOOLEAN,
+            "help_text": "Determines if Saleor should store cards on payments "
+            "in Stripe customer.",
+            "label": "Store customers card",
+        },
+        "automatic_payment_capture": {
+            "type": ConfigurationTypeField.BOOLEAN,
+            "help_text": "Determines if Saleor should automatically capture payments.",
+            "label": "Automatic payment capture",
+        },
+        "supported_currencies": {
+            "type": ConfigurationTypeField.STRING,
+            "help_text": "Determines currencies supported by gateway."
+            " Please enter currency codes separated by a comma.",
+            "label": "Supported currencies",
+        },
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        configuration = {item["name"]: item["value"] for item in self.configuration}
+        self.config = GatewayConfig(
+            gateway_name=GATEWAY_NAME,
+            auto_capture=configuration["automatic_payment_capture"],
+            supported_currencies=configuration["supported_currencies"],
+            connection_params={
+                "api_login_id": configuration["api_login_id"],
+                "transaction_key": configuration["transaction_key"],
+                "client_key": configuration["client_key"],
+                "use_sandbox": configuration["use_sandbox"],
+            },
+            store_customer=configuration["store_customers_card"],
+        )
+
+    def _get_gateway_config(self) -> GatewayConfig:
+        return self.config
+
+    @classmethod
+    def validate_plugin_configuration(cls, plugin_configuration: "PluginConfiguration"):
+        """Validate if provided configuration is correct."""
+        configuration = {
+            item["name"]: item["value"] for item in plugin_configuration.configuration
+        }
+
+        api_login_id = configuration.get("api_login_id", None)
+        transaction_key = configuration.get("transaction_key", None)
+
+        # Only check when both are set, otherwise the dashboard is hard to use
+        if api_login_id and transaction_key:
+            success, message = authenticate_test(
+                api_login_id, transaction_key, configuration.get("use_sandbox")
+            )
+            if not success:
+                raise ValidationError(
```

Comment:
The structure like this would be better:
```
                raise ValidationError(
                    {
                        "apple-pay-cert": ValidationError(
                            "The provided apple certificate is invalid.",
                            code=PluginErrorCode.INVALID.value,
                        )
                    }
                )
```
Where the `apple-pay-cert` is the defined field.
This is the standard structure of error which we use. I know there are some parts of code which the old error structure but the new part of code should use the proposed structure.

Reply:
@korycins as long as you spell Apple with a capital "A." 😄

Reply:
@korycins What about this case where it's at the form level and not a field level (because authentication failure is the result of 2 fields). In Django forms they refer to them as "non_field_errors". I could just pick on field and have it there.

Reply:
If the failure is related to more than one field you can still do something like this:
```
raise ValidationError(
                    {
                        "api-public-key": ValidationError(
                            "Provided keys are incorrect..",
                            code=PluginErrorCode.INVALID.value,
                        ),
                       "api-private-key": ValidationError(
                            "Provided keys are incorrect..",
                            code=PluginErrorCode.INVALID.value,
                        )
                    }
                )
```
The message is a little bit generic, but as a response, you return fields responsible for failure. 


You can still return the generic error if you don't know what fields are incorrect or missing.